src/azure-cli/azure/cli/command_modules/batch/_params.py

```diff
@@ -71,15 +82,46 @@ def load_arguments(self, _):
         c.argument('encryption_key_source', help='Part of the encryption configuration for the Batch account. Type of the key source. Can be either Microsoft.Batch or Microsoft.KeyVault', arg_type=get_enum_type(KeySource))
         c.argument('encryption_key_identifier', help='Part of the encryption configuration for the Batch account. '
                                                      'Full path to the versioned secret. Example https://mykeyvault.vault.azure.net/keys/testkey/6e34a81fef704045975661e297a4c053.')
-        c.argument('identity_type', help="The type of identity used for the Batch account. Possible values include: 'SystemAssigned', 'None'.", arg_type=get_enum_type(ResourceIdentityType))
+        c.argument('identity_type', deprecate_info=c.deprecate(hide=True))
+        c.argument('mi_user_assigned', help='Resource ID of the user assigned identity for the batch services account.', arg_group='Identity')
+        c.argument('mi_system_assigned', help='Set the system managed identity on the batch services account.', arg_group='Identity')
         c.ignore('keyvault_url')
 
     with self.argument_context('batch account set') as c:
         c.argument('tags', tags_type)
         c.argument('storage_account', help='The storage account name or resource ID to be used for auto storage.', validator=storage_account_id)
         c.argument('encryption_key_source', help='Part of the encryption configuration for the Batch account. Type of the key source. Can be either Microsoft.Batch or Microsoft.KeyVault')
+        c.argument('public_network_access', help="The network access type for accessing Azure Batch account. Values can either be enabled or disabled.", arg_type=get_enum_type(PublicNetworkAccessType))
         c.argument('encryption_key_identifier', help='Part of the encryption configuration for the Batch account. Full path to the versioned secret. Example https://mykeyvault.vault.azure.net/keys/testkey/6e34a81fef704045975661e297a4c053.')
-        c.argument('identity_type', help="The type of identity used for the Batch account. Possible values include: 'SystemAssigned', 'None'.", arg_type=get_enum_type(ResourceIdentityType))
+        c.argument('identity_type', deprecate_info=c.deprecate(hide=True))
+        c.argument('mi_system_assigned', help='Set the system managed identity on the batch services account.', arg_group='Identity')
+        c.argument('mi_user_assigned', help='Resource ID of the user assigned identity for the batch services account.', arg_group='Identity')
```

Comment:
For operating managed identity on existing resource, please refer to this part: [operate-managed-identity-on-existing-resource](https://github.com/Azure/azure-cli/blob/dev/doc/managed_identity_command_guideline.md#operate-managed-identity-on-existing-resource)

Reply:
updated

Reply:
> 1. Assign identities with `identity assign` command
> 2. Remove identities with `identity remove` command
> 3. Show identities with `identity show` command

Please note that there is a difference between operating managed identity on an existing resource and specifying a when creating a resource

Reply:
You may refer to this PR https://github.com/Azure/azure-cli/pull/22966 as an example

Reply:
added the identiy assign, remove, and show commands.  added a test as well